expr/func_movingwindow.go

```diff
@@ -93,25 +93,27 @@ func (s *FuncMovingWindow) Exec(dataMap DataMap) ([]models.Series, error) {
 	return outputs, nil
 }
 
-func aggregateOnWindow(serie models.Series, aggFunc batch.AggFunc, interval uint32, xFilesFactor float64) []schema.Point {
+func aggregateOnWindow(serie models.Series, aggFunc batch.AggFunc, interval uint32, xFilesFactor float64) (uint32, []schema.Point) {
 	out := pointSlicePool.Get().([]schema.Point)
 
 	numPoints := len(serie.Datapoints)
-	var serieStart, serieEnd uint32 = serie.QueryFrom, serie.QueryTo
+	serieStart, serieEnd := serie.QueryFrom, serie.QueryTo
 	queryStart := uint32(serieStart + interval)
 
-	// Find index of Point which corresponds to the start of the query
 	ptIdx := 0
-	for ptIdx < numPoints && serie.Datapoints[ptIdx].Ts <= queryStart {
+	for ptIdx < numPoints-1 {
+		if queryStart >= serie.Datapoints[ptIdx].Ts &&
```

Comment:
Is this check necessary? Since we are assuming time order, wouldn't it be sufficient to just check ptIdx+1? 
Also, is it possible in any scenario that our first point might be > queryStart? If so, this will loop will incorrectly forward past all points.

Reply:
I've rewritten it and added a check above for (queryStart<seriesStart) Please take a look.